interface/reports/daily_summary_report.php

```diff
@@ -0,0 +1,409 @@
+        if ($dateSet != 1 && $facilitySet != 1) {
+            $whereNewPatientConditions .= ' AND DATE(`OPE`.`pc_eventDate`) = ? ';
+            array_push($sqlBindArrayNewPatient, date("Y-m-d"));
+            $whereTotalVisitConditions .= ' AND DATE(`fc`.`date`) = ? ';
+            array_push($sqlBindArrayTotalVisit, date("Y-m-d"));
+            $whereTotalPaymentConditions .= ' AND DATE(`b`.`date`)  = ? ';
+            array_push($sqlBindArrayTotalPayment, date("Y-m-d"));
+            $wherePaidConditions .= ' AND DATE(`p`.`dtime`)  = ? ';
+            array_push($sqlBindArrayPaid, date("Y-m-d"));
+        }
+
+        // if search based on facility then append condition for facility search 
+        if (1 === $facilitySet) {
+            $facilityID = $selectedFacility;
+            $whereNewPatientConditions .= ' AND `f`.`id` = ?';
+            array_push($sqlBindArrayNewPatient, $selectedFacility);
+            $whereTotalVisitConditions .= ' AND `f`.`id` = ?';
+            array_push($sqlBindArrayTotalVisit, $selectedFacility);
+            $whereTotalPaymentConditions .= ' AND `f`.`id` = ?';
+            array_push($sqlBindArrayTotalPayment, $selectedFacility);
+            $wherePaidConditions .= ' AND `f`.`id` = ?';
+            array_push($sqlBindArrayPaid, $selectedFacility);
+        }
+
+        // if date range wise search then append condition for date search 
+        if (1 === $dateSet) {
+            $whereNewPatientConditions .= ' AND DATE(`OPE`.`pc_eventDate`) BETWEEN ? AND ?';
+            array_push($sqlBindArrayNewPatient, $selectedFromDate, $selectedToDate);
+            $whereTotalVisitConditions .= ' AND DATE(`fc`.`date`) BETWEEN ? AND ?';
+            array_push($sqlBindArrayTotalVisit, $selectedFromDate, $selectedToDate);
+            $whereTotalPaymentConditions .= ' AND DATE(`b`.`date`) BETWEEN ? AND ?';
+            array_push($sqlBindArrayTotalPayment, $selectedFromDate, $selectedToDate);
+            $wherePaidConditions .= ' AND DATE(`p`.`dtime`) BETWEEN ? AND ?';
+            array_push($sqlBindArrayPaid, $selectedFromDate, $selectedToDate);
+        }
+
+        // if provider selected then append condition for provider
+        if (isset($selectedProvider) && !empty($selectedProvider)) {
+            $providerID = $selectedProvider;
+            $whereNewPatientConditions .= ' AND `OPE`.`pc_aid` = ?';
+            array_push($sqlBindArrayNewPatient, $selectedProvider);
+            $whereTotalVisitConditions .= ' AND `fc`.`provider_id` = ?';
+            array_push($sqlBindArrayTotalVisit, $selectedProvider);
+            $whereTotalPaymentConditions .= ' AND `fe`.`provider_id` = ?';
+            array_push($sqlBindArrayTotalPayment, $selectedProvider);
+            $wherePaidConditions .= ' AND `fe`.`provider_id` = ?';
+            array_push($sqlBindArrayPaid, $selectedProvider);
+        }
+
+        // pass last parameter as Boolean,  which is getting the facility name in the resulted array 
```

Comment:
I agree with your comment. However we need to use last boolean parameter for the following two purpose.

1. To get facility name for the appointment. This can be done via simply run  a separate query to get facility name for appointment but this will add extra "for loop" to get it and impact on the report performance. So we thought to just include the facility name while fetching all appointment.

2. To bypass following condition where we don't require to check the facility id in the user table because providers may belongs to multiple facilities and have appointments from the multiple facilities. so if we don't exclude 'u.facility_id = ?' condition then it is not showing result for the selected facilities.

We have also noticed that appointments report under "reports --> visits" is also not showing appointment for the specific facility selected even there is an appointment booked for that facility.

  $where .= " AND e.pc_facility = ? AND u.facility_id = ?";